driver-core/src/main/java/com/datastax/driver/core/MaterializedViewMetadata.java

```diff
@@ -178,21 +172,12 @@ public String apply(ColumnMetadata input) {
         newLine(sb.append("FROM ").append(keyspaceName).append('.').append(baseTableName).append(" "), formatted);
 
         // WHERE
-        newLine(sb.append("WHERE "), formatted);
-        Iterator<ColumnMetadata> it = getPrimaryKey().iterator();
-        while(it.hasNext()) {
-            ColumnMetadata column = it.next();
-            sb.append(spaces(4, formatted)).append(Metadata.escapeId(column.getName()));
-            sb.append(" IS NOT NULL");
-            if(it.hasNext()) sb.append(" AND");
-            sb.append(" ");
-            newLine(sb, formatted);
-        }
+        newLine(sb.append("WHERE ").append(whereClause).append(' '), formatted);
```

Comment:
Got an [explanation](https://issues.apache.org/jira/browse/CASSANDRA-6477?focusedCommentId=14627173&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14627173) that for now the where clause will be required for now, but it does seem possible in the future that this could be optional, so I think it would still be good to handle this case.